tests/linux_kernel/test_stack_trace.py

```diff
@@ -59,6 +62,60 @@ def test_by_pid_dwarf(self):
     def test_by_pid_orc(self):
         self._test_by_pid(True)
 
+    def _check_logged_orc_message(self, captured_logs, module):
+        # To be sure that we actually used ORC to unwind through the drgn_test
+        # stack frames, search for the log output. We don't know which ORC
+        # version is used, so just ensure that we have a log line that mentions
+        # loading ORC.
+        expr = re.compile(
+            r"DEBUG:drgn:Loaded built-in ORC \(v\d+\) for module " + module
+        )
+        for line in captured_logs.output:
+            if expr.fullmatch(line):
+                break
+        else:
+            self.fail(f"Did not load built-in ORC for {module}")
+
+    @unittest.skipUnless(
+        NORMALIZED_MACHINE_NAME == "x86_64",
+        f"{NORMALIZED_MACHINE_NAME} does not use ORC",
+    )
+    @skip_unless_have_test_kmod
+    def test_by_pid_builtin_orc(self):
+        # ORC was introduced in kernel 4.14. Detect the presence of ORC or skip
+        # the test.
+        try:
+            self.prog.symbol("__start_orc_unwind")
+        except LookupError:
+            ver = self.prog["UTS_RELEASE"].string_().decode()
+            self.skipTest(f"ORC is not available for {ver}")
+
+        with drgn_log_level(logging.DEBUG):
+            # Create a program with the core kernel debuginfo loaded,
+            # but without module debuginfo. Load a symbol finder using
+            # kallsyms so that the module's stack traces can still have
+            # usable frame names.
+            prog = Program()
+            prog.set_kernel()
+            try:
+                prog.load_default_debug_info()
+            except MissingDebugInfoError:
+                pass
```

Comment:
Do you need to load the debug info for all of the other modules? If not, how about
```suggestion
            prog.load_debug_info(main=True)
```

Reply:
Drgn needs to at least _enumerate_ the modules, so that there is a `drgn_module` created. Otherwise, we have nowhere to put the ORC and the test fails. But we don't need to go searching for the debuginfo. I have achieved this in other places by placing a dummy call of `list(prog.loaded_modules())`.

Reply:
I went ahead and used `load_debug_info(main=True)` and added a call to `list(prog.loaded_modules())`. Though I'm not sure that is 100% satisfactory so I'll leave this open for further discussion.